src/redux/books/books.js

```diff
@@ -1,52 +1,77 @@
-// ACTIONS
-const ADD_BOOK = 'bookstore/books/ADD';
-const REMOVE_BOOK = 'bookstore/books/REMOVE';
+/* eslint-disable no-param-reassign */
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/8zgJXl3G9vojkee6XZ5H/books';
 
 // INITIAL STATE
-export const initialState = [
-  {
-    id: 1,
-    title: 'Lean Redux',
-    author: 'Microverse',
-  },
-  {
-    id: 2,
-    title: 'Lean Advanced Redux',
-    author: 'FreeCodeCamp',
-  },
-  {
-    id: 3,
-    title: 'Learn ReduxToolkit',
-    author: 'Code Cademy',
+const initialState = {
+  books: [],
+  isLoading: false,
+};
+
+export const getBooks = createAsyncThunk('books/getbooks', async () => {
+  const response = await fetch(url);
+  const data = await response.json();
+  const books = [Object.keys(data).map((key) => (
+    {
+      id: key,
+      ...data[key][0],
+    }
+  ))];
+  return books;
+});
+
+export const addBook = createAsyncThunk(
+  'books/addBook', async (payload, thunkAPI) => {
+    await fetch(`${url}`, {
+      method: 'POST',
+      body: JSON.stringify({
+        item_id: payload.id,
+        title: payload.title,
+        author: payload.author,
+        category: payload.category,
+      }),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    }).then(() => thunkAPI.dispatch(getBooks()));
+    const { books } = thunkAPI.getState().books;
+    return books;
   },
-];
+);
 
-// ACTION CREATORS
-export function addAction(book) {
-  return {
-    type: ADD_BOOK,
-    book,
-  };
-}
+export const removeBook = createAsyncThunk(
+  'books/removeBook', async (payload, thunkAPI) => {
+    await fetch(`${url}/${payload.id}`, {
+      method: 'DELETE',
+    }).then(() => thunkAPI.dispatch(getBooks()));
+    const { books } = thunkAPI.getState().books;
+    return books;
+  },
+);
 
-export function removeAction(id) {
-  return {
-    type: REMOVE_BOOK,
-    payload: id,
-  };
-}
+const bookSlice = createSlice({
+  name: 'book',
+  initialState,
+  reducers: {},
+  extraReducers: {
+    [getBooks.pending]: (state) => {
+      state.isLoading = true;
+    },
+    [getBooks.fulfilled]: (state, action) => {
+      state.isLoading = false;
+      state.books = action.payload[0]; //eslint-disable-line
```

Comment:
- [ ] Kindly avoid disabling linters as it is not a good practice, it is better for your linters to pass the linter checks 👍🏿

Reply:
Hello @prolajumokeoni when I use ReduxToolKit linters are showing errors because of the syntax. so is there anything I can do to avoid this?

Reply:
- [ ]  Kindly share the error message you get, you can also google the error message to see how others have fixed it, You can also share on the slack students channel 😉